zipline/utils/events.py

```diff
@@ -353,6 +357,13 @@ def calculate_dates(self, dt):
             self.cal.open_and_close_for_session(
                 self.cal.minute_to_session_label(dt)
             )
+
+        # When using the futures calendar, adjust the open and close times to
+        # conform to 6:30am to 5:00pm.
+        if self.cal.name == 'us_futures':
```

Comment:
How about we add `execution_open` and `execution_close` to the base calendar API. By default it's just the regular open and close, but a calendar can override to provide custom functionality (requiring that execution open is after the regular open, etc).

Then you can just get all the execution opens by asking the calendar.